spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/types/external.scala

```diff
@@ -108,23 +108,32 @@ object external {
     }
   }
 
+  private def qualifiedPath(path: String): Path = {
+    val p = new Path(path)
+    val fs = p.getFileSystem(SparkUtils.sparkSession.sparkContext.hadoopConfiguration)
+    p.makeQualified(fs.getUri, fs.getWorkingDirectory)
+  }
+
   private def resolveURI(path: String): URI = {
-    try {
-      val uri = new URI(path)
-      if (uri.getScheme() != null) {
-        return uri
-      }
+    val uri = new URI(path)
+    if (uri.getScheme() != null) {
+      return uri
+    }
+
+    val qUri = qualifiedPath(path).toUri
+    if (qUri.getScheme == "file") {
```

Comment:
why this additional check ?

Reply:
This is from existing code (it looks to be borrowed from Spark) and I wouldn't want to drop the existing code where I'm not 100% sure.

Reply:
I also have the same question that why is this additional check added?

Reply:
Because below code only makes sense for local file, as it retrieves absolute path via leveraging `File`. For HDFS file, we may not even need to do it, as we expect absolute path for HDFS once it qualifies.

That said, if `makeQualified` guarantees to always convert the path as absolute path, we even don't need this. I'll check it.

Reply:
`makeQualified` looks to convert the path as absolute path, but scheme part looks to be different.

If we apply below logic, `file://` is used as a prefix before path. If we rely on `makeQualified`, `file` is used.

From reference of [wikipedia](https://en.wikipedia.org/wiki/File_URI_scheme), both seem to be valid file URI, but given we have been providing it as `file://`, we might be better to just leave this as it is.

Reply:
Can you add this in code comments?

Reply:
OK. I'll add the comment.

Reply:
Updated.